lib/stateView/document/query/errors/NotIndexedFieldError.js

```diff
@@ -0,0 +1,9 @@
+const ValidationError = require('./ValidationError');
+
+class NotIndexedFieldError extends ValidationError {
+  constructor(fields) {
+    super(`Search fields can only contain one of these fields: ${fields.join(', ')}`);
+  }
+}
```

Comment:
Let's add getter in case if we want to check fields programmatically: `getIndexedFields()`